### PR TITLE
fix(complete): nudge interactive+no_confirm (-y) mode on think-only responses

### LINKED ISSUE #2725: bug: LLM produces thinking but no tool output, causing silent re-prompts in non-interactive mode
## Problem

Sometimes the LLM generates a response that contains only "thinking" (reasoning/planning text) but no tool use blocks. In `--non-interactive` mode, the `complete` hook detects "no tool was executed" and re-prompts the assistant to continue — which can cause:

1. **Wasted inference**: the thinking tokens are paid for but produce no action
2. **Silent retry loops**: the re-prompt may trigger the same pattern again
3. **Hard to diagnose**: there's no obvious WARNING log when this happens, so it's easy to miss in production autonomous runs

## Observed behavior

In an interactive session, the assistant produced several messages where tool blocks were present in the response formatting but never actually executed (the assistant appeared to be "thinking" about what to do without committing to action).

## Investigation needed

- [ ] Grep gptme logs for the re-prompt pattern in `--non-interactive` mode — how often does the "no tool executed, continuing" path trigger?
- [ ] Check if this correlates with specific models (e.g. reasoning models like o1/o3 or extended thinking in Claude)
- [ ] Consider adding a WARNING log when the assistant produces a response with no executable tool use in non-interactive mode, so it's visible in log output
- [ ] Track the count of consecutive no-tool-use responses and circuit-break if it loops

## For Bob

Bob should:
1. **Grep his own session logs** for cases where `--non-interactive` re-prompted due to no tool use — this would show up in the chat loop logic
2. **Add WARNING log grepping to his health checks/vitals** — a general scan for WARNING-level log lines across recent sessions would surface this and other silent issues
3. **Check trajectories** for patterns where the assistant produces long reasoning blocks but no actionable output, especially in autonomous runs where this wastes budget silently

## Suggested improvements

1. Add `logger.warning("Assistant produced no tool use in non-interactive mode, re-prompting")` at the re-prompt point
2. Consider a metric/counter for no-tool-use responses per session
3. Add a vitals/health-check entry that scans recent logs for WARNING lines as a general early-warning system

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Problem

When running `gptme -y` (interactive mode with auto-confirmation), a think-only LLM response (text/reasoning without any tool calls) silently returns control to the terminal. The user must manually type "continue" to prod the assistant into action.

In non-interactive mode, `auto_reply_hook` already handles this by auto-injecting a nudge + exit after 2 retries. But the interactive path early-returns because `no_confirm` wasn't threaded into the LOOP_CONTINUE hook — the hook can't distinguish `-y` mode from a real human conversation.

## Fix

Minimal two-part change:

1. **Thread `no_confirm` through `chat.py` → `_run_chat_loop` → LOOP_CONTINUE trigger** so the hook receives it
2. **Extend `auto_reply_hook`** to handle `interactive=True, no_confirm=True` with a single quiet nudge (no exit path — unlike non-interactive mode's 2x-then-exit)

Also adds `no_confirm` to the `LoopContinueHook` protocol and updates `stuck_detect_hook` / `_subagent_completion_hook` to match.

## Testing

4 new tests cover the interactive+no_confirm behavior:
- Nudges on think-only response
- No nudge when tools are present
- Only nudges once per think-only sequence
- No exit logic (no SessionCompleteException escalation)

All 54 existing tests in `test_tools_complete.py` continue to pass.

Closes #2725

### CODE REVIEW

**Review by TimeToBuildBob — COMMENTED** [review]
Thanks for the review @greptile-apps!

- **prompt_queue guard**: ✅ Applied — added `if not prompt_queue` before the nudge in the interactive+no_confirm path (d819952)
- **Docstring**: ✅ Fixed — corrected "once per conversation" to "once per uninterrupted think-only sequence" (d819952)
- **Architectural concern about no_confirm**: The alternative would be to deduplicate the `interactive=True, no_confirm=True` case through the existing `interactive=True` path with a broader check for "-y mode", but that would require the hook to detect no_confirm from the manager state (which isn't readily available). The explicit `no_confirm` parameter is the least invasive way to thread the signal through the hook chain. Happy to explore alternatives if you have a concrete suggestion for keeping the Protocol surface smaller.

CI typecheck failure is also fixed (mypy Protocol/@overload resolution issue in test_confirm.py).

**Comment by TimeToBuildBob** [comment]
## Coverage improvement

Added two new tests covering edge cases in `_auto_reply_nudge_interactive`:

1. **`test_interactive_no_confirm_noop_when_no_assistant_msg`** — covers the early return when the log has no assistant messages (previously uncovered line).
2. **`test_interactive_no_confirm_nudge_stops_at_prior_tool_use`** — covers the nudge counting loop boundary when it encounters a prior assistant turn with tools (previously uncovered line).

Overall coverage of `gptme/tools/complete.py` improves from 94% → 96%. The remaining uncovered lines are pre-existing (not part of this PR's patch).

All 56 tests pass.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/tools/complete.py [hunk 1]
```diff
@@ -97,26 +97,87 @@ def complete_hook(
...
-    if interactive:
+    # In interactive mode without -y, skip (real human conversation)
+    if interactive and not no_confirm:
+        return
+
+    # In interactive + no_confirm mode: gentle nudge, no exit path
+    if interactive and no_confirm:
+        if not prompt_queue:
+            yield from _auto_reply_nudge_interactive(manager)
         return
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/tools/complete.py:177]: Fixed in `f424de572`.

The hook dispatcher now filters trigger kwargs against each hook callable's signature before invocation. Hooks with `**kwargs` still receive the full context, while older fixed-signature hooks simply don't receive newly added kwargs like `no_confirm`, so existing third-party `LoopContinueHook` implementations keep working.

Added a regression test that registers a legacy three-argument `LOOP_CONTINUE` hook and triggers it with `no_confirm=True`.

Verification:
- `pytest tests/test_hooks.py tests/test_tools_complete.py` — 73 passed
- targeted `mypy` on the touched hook/complete/chat files — clean
- `ruff check gptme/hooks/registry.py tests/test_hooks.py` — clean
- `prek run --files gptme/hooks/registry.py tests/test_hooks.py` — passed
